inst/include/dust/distr/poisson.hpp

```diff
@@ -102,7 +103,7 @@ int rpois(rng_state_t<real_t>& rng_state,
       // The expression below is equivalent to the computation of step 2)
       // in transformed rejection (v <= alpha * F'(G(u)) * G'(u)).
       real_t s = std::log(v * inv_alpha / (a / (u_shifted * u_shifted) + b));
-      real_t t = -lambda + k * log_rate - std::lgamma(k + 1);
+      real_t t = -lambda + k * log_rate - dust::utils::lgamma<real_t>(k + 1);
```

Comment:
Would it make sense to update the lgamma calls in the densities now too? Happy for this to be left for a future PR also